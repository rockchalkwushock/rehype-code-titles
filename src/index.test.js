import { rehype } from 'rehype'
import dedent from 'dedent'
import rehypeCodeTitles from '../index.js'

const processHtml = html => {
  return rehype()
    .data('settings', { fragment: true })
    .use(rehypeCodeTitles)
    .processSync(html)
    .toString()
}

const classNameWithTitle = 'language-typescript:lib/mdx.ts'
const className = 'language-typescript'

describe('rehype-code-titles', () => {
  test('Snapshot w/ title block', () => {
    const result = processHtml(dedent`
      <pre><code class="${classNameWithTitle}"></code></pre>
    `)
    expect(result).toMatchSnapshot()
  })
  test('Snapshot w/out title block', () => {
    const result = processHtml(dedent`
      <pre><code class="${className}"></code></pre>
    `)
    expect(result).toMatchSnapshot()
  })
  test('Contains title block', () => {
    const result = processHtml(dedent`
      <pre><code class="${classNameWithTitle}"></code></pre>
    `)
    expect(result.includes('div')).toEqual(true)
    expect(result.includes('rehype-code-title')).toEqual(true)
    expect(result.includes('lib/mdx.ts')).toEqual(true)
  })
  test('Does not contain title block', () => {
    const result = processHtml(dedent`
      <pre><code class="${className}"></code></pre>
    `)
    expect(result.includes('div')).not.toEqual(true)
    expect(result.includes('rehype-code-title')).not.toEqual(true)
    expect(result.includes('lib/mdx.ts')).not.toEqual(true)
  })
})
