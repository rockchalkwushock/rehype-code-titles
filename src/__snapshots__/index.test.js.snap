// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rehype-code-titles Snapshot w/ title block 1`] = `"<div class=\\"rehype-code-title\\">lib/mdx.ts</div><pre><code class=\\"language-typescript\\"></code></pre>"`;

exports[`rehype-code-titles Snapshot w/out title block 1`] = `"<pre><code class=\\"language-typescript\\"></code></pre>"`;
